newIDE/app/src/UI/SemiControlledTextField.js

```diff
@@ -121,7 +144,17 @@ export default class SemiControlledTextField extends React.Component<
           if (!commitOnBlur) onChange(newValue);
         }}
         onBlur={event => {
-          onChange(event.currentTarget.value);
+          let value = event.currentTarget.value;
+          if (this.props.calculValue) {
+            const rex = /(^\d).*[-+*()/.,].*(\d+$)|(^\d+$)|^((\d)|[-+*()/.,]).((\d)|[-+*()/.,])+$/gm;
```

Comment:
You are not escaping the slashes, are you sure this is working?
```suggestion
            const rex = /(^\d).*[-+*()\/.,].*(\d+$)|(^\d+$)|^((\d)|[-+*()\/.,]).((\d)|[-+*()\/.,])+$/gm;
```